sensor/kubernetes/certrefresh/securedcluster/certificate_requester.go

```diff
@@ -0,0 +1,148 @@
+// as a call to RequestCertificates.
+func (r *certificateRequesterImpl) Start() {
+	r.stopC.Reset()
+	go r.dispatchResponses()
+}
+
+// Stop makes the certificate stop forwarding responses to running requests. Subsequent calls to RequestCertificates
+// will fail with ErrCertificateRequesterStopped.
+// Currently active calls to RequestCertificates will continue running until cancelled or timed out via the
+// provided context.
+func (r *certificateRequesterImpl) Stop() {
+	r.stopC.Signal()
+}
+
+func (r *certificateRequesterImpl) dispatchResponses() {
+	for {
+		select {
+		case <-r.stopC.Done():
+			return
+		case msg := <-r.receiveC:
+			responseC, ok := r.requests.Load(msg.GetRequestId())
+			if !ok {
+				log.Debugf("request ID %q does not match any known request ID, dropping response",
+					msg.GetRequestId())
+				continue
+			}
+			r.requests.Delete(msg.GetRequestId())
+			// Doesn't block even if the corresponding call to RequestCertificates is cancelled and no one
+			// ever reads this, because requestC has buffer of 1, and we removed it from `r.request` above,
+			// in case we get more than 1 response for `msg.GetRequestId()`.
+			responseC.(chan *central.IssueSecuredClusterCertsResponse) <- msg
+		}
+	}
+}
+
+// RequestCertificates makes a new request for a new set of secured cluster certificates from Central.
+// This assumes the certificate requester is started, otherwise this returns ErrCertificateRequesterStopped.
+func (r *certificateRequesterImpl) RequestCertificates(ctx context.Context) (*certificates.Response, error) {
+	// Central capabilities are only available after this component is created,
+	// which is why this check is done here
+	if !centralcaps.Has(centralsensor.SecuredClusterCertificatesReissue) {
+		return nil, errors.New("TLS certificate refresh failed: missing Central capability 'SecuredClusterCertificatesReissue'")
+	}
+
+	requestID := uuid.NewV4().String()
+	receiveC := make(chan *central.IssueSecuredClusterCertsResponse, 1)
+	r.requests.Store(requestID, receiveC)
+	// Always delete this entry when leaving this scope to account for requests that are never responded, to avoid
+	// having entries in `r.requests` that are never removed.
+	defer r.requests.Delete(requestID)
+
+	if err := r.send(ctx, requestID); err != nil {
+		return nil, err
+	}
+	return receive(ctx, receiveC)
+}
+
+func (r *certificateRequesterImpl) send(ctx context.Context, requestID string) error {
+	msg := &central.MsgFromSensor{
+		Msg: &central.MsgFromSensor_IssueSecuredClusterCertsRequest{
+			IssueSecuredClusterCertsRequest: &central.IssueSecuredClusterCertsRequest{
+				RequestId: requestID,
+			},
+		},
+	}
+	select {
+	case <-r.stopC.Done():
+		return r.stopC.ErrorWithDefault(ErrCertificateRequesterStopped)
+	case <-ctx.Done():
+		return ctx.Err()
+	case r.sendC <- message.New(msg):
+		return nil
+	}
+}
+
+func convertToIssueCertsResponse(response *central.IssueSecuredClusterCertsResponse) *certificates.Response {
```

Comment:
Why not a function in the `sensor/kubernetes/certrefresh/certificates` package, like there is for `localscanner/certificate_requester.go`? Perhaps the local scanner function is a bit of an overkill, but I think in this case consistency is more important.

Reply:
Good catch, I did the function in the `certificates` package as a review suggestion in a previous PR, and this function was written before that - and then I forgot about it.